src/components/Listbox/Listbox.tsx

```diff
@@ -0,0 +1,23 @@
+import { Listbox as HeadlessListbox } from "@headlessui/react";
+import { PropsWithChildren } from "react";
+
+export interface ListboxProps<T> extends PropsWithChildren {
+  value: T;
+  onChange: (value: T) => void;
+  className?: string;
+  /** If you provide `multiple`, then `value` and `onChange` must use an array. */
+  multiple?: boolean;
+}
+
+export const Listbox = <T extends object>({
+  children,
+  value,
+  onChange,
+  multiple,
+}: ListboxProps<T>) => {
+  return (
+    <HeadlessListbox multiple={multiple} value={value} onChange={onChange}>
```

Comment:
Instead of having `multiple` being a prop, could we check the type `T` of value with `isArray`?